lib/Onyx.js

```diff
@@ -675,18 +675,25 @@ function mergeCollection(collectionKey, collection) {
             const keyValuePairsForExistingCollection = prepareKeyValuePairsForStorage(existingKeyCollection);
             const keyValuePairsForNewCollection = prepareKeyValuePairsForStorage(newCollection);
 
+            const promises = [];
+
             // New keys will be added via multiSet while existing keys will be updated using multiMerge
             // This is because setting a key that doesn't exist yet with multiMerge will throw errors
-            const existingCollectionPromise = AsyncStorage.multiMerge(keyValuePairsForExistingCollection);
-            const newCollectionPromise = AsyncStorage.multiSet(keyValuePairsForNewCollection);
+            if (keyValuePairsForExistingCollection.length > 0) {
+                promises.push(AsyncStorage.multiMerge(keyValuePairsForExistingCollection));
+            }
+
+            if (keyValuePairsForNewCollection.length > 0) {
+                promises.push(AsyncStorage.multiSet(keyValuePairsForNewCollection));
+            }
```

Comment:
After the update `AsyncStorage.multiSet` throws when it is called with an empty array